reference/7.2/Microsoft.PowerShell.Core/About/about_Environment_Variables.md

```diff
@@ -308,6 +311,16 @@ System Control Panel:
 1. Select **Environment Variables...**.
 1. Make your changes.
 
+## Create persistent environment variables on non-Windows platforms
+
+Linux and macOS have configuration files and scripts that the operating system
+uses to set environment variables before starting an application. For example
+on Linux, you can add environment variables to the `/etc/environment` file or
+create a script that sets environment variables and put it in the
+`/etc/profile.d` folder. On macOS, you can add environment variables to the
+`/etc/zshrc` file. For more information, see the documentation for your
```

Comment:
Note that `/etc/zshrc` is the appropriate file to use only if `zsh` is a user's default shell.

If _PowerShell_ is the default shell, `/etc/profile` must be used, because PowerShell, when invoked with `-l` (as happens implicitly on macOS for the default shell), it reinvokes itself via `zsh -l` with the latter in POSIX compatibility mode, where only `/etc/profile` is sourced.

Reply:
@mklement0, can you point to documentation for `zsh` that states that. The documentation I found indicated that `/etc/zshrc` would be invoked by `zsh -l`.

Reply:
As also noted in the linked issue, the `/etc/profile` / `/etc/profile.d` approaches on non-GUI Linux distros - e.g. in Docker images - require explicit invocation via `pwsh -l` to take effect.